pypesto/petab/importer.py

```diff
@@ -496,32 +513,27 @@ def create_objective(
                 edatas,
                 inner_options,
             )
-            amici_reporting = amici.RDataReporting.residuals
-
-        elif self._hierarchical:
-            inner_problem = InnerProblem.from_petab_amici(
-                self.petab_problem, model, edatas
-            )
-            calculator = HierarchicalAmiciCalculator(inner_problem)
             amici_reporting = amici.RDataReporting.full
 
-        if self._hierarchical:
             # FIXME: currently not supported with hierarchical
             if 'guess_steadystate' in kwargs and kwargs['guess_steadystate']:
                 warnings.warn(
                     "`guess_steadystate` not supported with hierarchical optimization. Disabling `guess_steadystate`."
                 )
             kwargs['guess_steadystate'] = False
-            inner_parameter_ids = calculator.get_inner_parameter_ids()
+            inner_parameter_ids = calculator.get_inner_par_ids()
             par_ids = [x for x in par_ids if x not in inner_parameter_ids]
 
         max_sensi_order = kwargs.pop('max_sensi_order', None)
-        if self._non_quantitative_data_types:
+        if self._non_quantitative_data_types is not None and any(
```

Comment:
todo could be a bit more verbose

Reply:
Should the FIXME be converted into an issue?!

Reply:
> todo could be a bit more verbose
Should I change the `max_sensi_order` error to `raise ValueError ...`?

> Should the FIXME be converted into an issue?!
The FIXME has been here for quite some time already. But I can create an enhancement issue about it. But I don't think I have heard any wishes for it to be implemented yet.